src/Modal.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+
+import React from 'react';
+import classNames from 'classnames';
+import './Modal.scss';
+
+type ModalProps = {
+    show: boolean,
+    focusElementSelector: string,
```

Comment:
I think rename `focusElementSelector` to `focusOnOpenSelector` to match `focusOnCloseSelector` and to make the role clearer.